WordPress/gutenberg
# wp-env https support 8211

## What?
Provides HTTPS support for wp-env by allows to define the ports used for HTTPS, path of the certificate files, and even creates certificate files if none are specified.

## Why?

Addresses #8211
Old PR #53959

> For another use case, I'm currently developing a plugin for CSP management, and Chromium browsers require HTTPS for the `report-to` directive.

I'm sure there are other scenarios that require the use of HTTPS. This allows inexperienced users to easily enable HTTPS if necessary.

## How?

If `https` is enabled in the `.wp-env.json` config, it uses the default or user defined ports and certificate keypair to enable. You can define a certificate keypair to be used or let `devcert` create certificates if none are defined. A new apache config is created and enabled using this cert. The the `WP_HOME` and `WP_SITEURL` constants are updated as well.

## Testing Instructions





1. Add `"https": true` to `.wp-env.json` or `.wp-env.override.json`
2. Run `wp-env start`

__To define ports:__
Add `"httpsPort": 1234` and/or `"testsHttpsPort": 5678` to the config root.

__To define certificate files:__
Add `"sslCertPath": "/path/to/file.crt"` and `"sslKeyPath": "/path/to/file.key"` to the config root. Both must be defined.

Note that because of my limited experience, I have not tested with xdebug and the like. However, both the development and tests instances uses the certificate as expected.
